src/Logging/TeamCity/Converter.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pest\Logging\TeamCity;
+
+use Pest\Exceptions\ShouldNotHappen;
+use Pest\Support\Str;
+use PHPUnit\Event\Code\Test;
+use PHPUnit\Event\Code\TestMethod;
+use PHPUnit\Event\Code\Throwable;
+use PHPUnit\Event\TestSuite\TestSuite;
+use PHPUnit\Framework\Exception as FrameworkException;
+
+/**
+ * @internal
+ */
+final class Converter
+{
+    public function __construct(
+        private readonly string $rootPath,
+    )
+    {
+    }
+
+    public function getTestCaseMethodName(Test $test): string
+    {
+        if (! $test instanceof TestMethod) {
+            throw ShouldNotHappen::fromMessage("Not an instance of TestMethod");
+        }
+
+        return $test->testDox()->prettifiedMethodName();
+    }
+
+    public function getTestCaseLocation(Test $test): string
+    {
+        if (! $test instanceof TestMethod) {
+            throw ShouldNotHappen::fromMessage("Not an instance of TestMethod");
+        }
+
+        $fileName = $test->testDox()->prettifiedClassName();
+        // TODO: Get the description without the dataset.
```

Comment:
@nunomaduro could use your help on this one to figure out how we can get this information in pest 2.x

Reply:
@olivernybroe To order to help you the best I can, can you please upload a failing tests if possible?

Reply:
yeah, haven't worked really on the testing part for all of this. Will see what I can do